src/options/parser.rs

```diff
@@ -1,815 +1,248 @@
+    pub time: Option<OsString>,
+    /// use the accessed timestamp field.
+    #[arg(short = 'u', long, action = clap::ArgAction::Count)]
+    pub accessed: u8,
+    /// use the created timestamp field.
+    #[arg(short = 'U', long, action = clap::ArgAction::Count)]
+    pub created: u8,
+    /// how to format timestamps (default, iso, long-iso, full-iso, relative).
+    #[arg(long = "time-style")]
+    pub time_style: Option<OsString>,
+    /// display entries as hyperlinks.
+    #[arg(long, action = clap::ArgAction::Count)]
+    pub hyperlink: u8,
+    /// suppress the permissions field.
+    #[arg(long = "no-permissions", action = clap::ArgAction::Count)]
+    pub no_permissions: u8,
+    /// suppress the filesize field.
+    #[arg(long = "no-filesize", action = clap::ArgAction::Count)]
+    pub no_filesize: u8,
+    /// suppress the user field.
+    #[arg(long = "no-user", action = clap::ArgAction::Count)]
+    pub no_user: u8,
+    /// suppress the time field.
+    #[arg(long = "no-time", action = clap::ArgAction::Count)]
+    pub no_time: u8,
+    /// don't display icons (always override --icons).
+    #[arg(long = "no-icons", action = clap::ArgAction::Count)]
+    pub no_icons: u8,
+    /// suppress git.
+    #[arg(long = "no-git", action = clap::ArgAction::Count)]
+    pub no_git: u8,
+    /// list root of git-tree status.
+    #[arg(long = "git-repos", action = clap::ArgAction::Count)]
+    pub git_repos: u8,
+    ///List each git-repos branch name (much faster)
+    #[arg(long = "git-repos-no-status", action = clap::ArgAction::Count)]
+    pub git_repos_no_status: u8,
+    /// list each file's permission in octal format.
+    #[arg(short = 'o', long, alias = "octal-permission", alias = "octal-permissions", action = clap::ArgAction::Count)]
+    pub octal: u8,
+    /// Display the number of hard links to file.
+    #[arg(short = 'Z', long = "context", action = clap::ArgAction::Count)]
+    pub security_context: u8,
+    /// Show extended attributes.
+    #[arg(short = '@', long, action = clap::ArgAction::Count)]
+    pub extended: u8,
+    /// Show list of command-line options.
+    #[arg(short ='?', long, action = clap::ArgAction::Help)]
+    pub help: Option<bool>,
+    /// Show mount details (Linux only)
+    #[arg(short = 'M', long, action = clap::ArgAction::Count)]
+    pub mounts: u8,
+    /// Show only files
+    #[arg(short = 'f', long = "only-files", action = clap::ArgAction::Count)]
+    pub only_files: u8,
+    /// Don't Show quotes
+    #[arg(long = "no-quotes", action = clap::ArgAction::Count)]
+    pub no_quotes: u8,
+    /// only show group if it has a different name from owner
+    #[arg(long = "smart-group", action = clap::ArgAction::Count)]
+    pub smart_group: u8,
+    /// show the size of a directory as the size of all files and directories inside
+    #[arg(long = "total-size", action = clap::ArgAction::Count)]
+    pub total_size: u8,
+}
+
+impl Opts {
+    pub fn default() -> Opts {
```

Comment:
It seems strange to derive `Default` and implement a `default` method outside of `Default`.

Reply:
Well caught